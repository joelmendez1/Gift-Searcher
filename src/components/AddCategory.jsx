import { useState } from 'react';
import PropTypes from "prop-types"

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = ({ target }) => {
    setInputValue(target.value);
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    const newInputValue = inputValue.trim();

    if(newInputValue.length <= 1) return;

    onNewCategory(newInputValue);
    setInputValue('');
  }

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input 
      type='text' 
      placeholder='Buscar gifs'
      value={inputValue}
      onChange={handleInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
