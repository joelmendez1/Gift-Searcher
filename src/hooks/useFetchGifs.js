import { useEffect, useState } from "react";
import { getGifs } from "../utils/getGifs";

export const useFetchGifs = (category) => {
  const [images, setImages] = useState([]);
  const [isLoading, setIsLoading] = useState(true)

  useEffect(() => {
    getGifs(category)
      .then(res => setImages(res))
      .then(() => setIsLoading(false))
  }, [category])

  return {
    images,
    isLoading
  }
}
