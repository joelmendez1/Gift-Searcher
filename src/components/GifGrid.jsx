import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './index';
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category)

  return (
    !isLoading ?
    (
      <>
        <h3>{category}</h3>
        <div className='card-grid'>
          {
            images.map((image) => {
              return <GifItem key={image.id} {...image}/>
            })
          }
        </div>
      </>
    ) : <h2>Gifs cargando...</h2>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
