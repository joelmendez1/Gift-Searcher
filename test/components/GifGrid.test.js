import { render, screen } from "@testing-library/react";
import { GifGrid } from "../../src/components/GifGrid";
import { useFetchGifs } from "../../src/hooks/useFetchGifs";

jest.mock('../../src/hooks/useFetchGifs');

const gifs = [
  {
    id: 1,
    title: "One Punch",
    url: "Mock URL OP"
  },
  {
    id: 2,
    title: "Dragon Ball",
    url: "Mock URL DB"
  },
  {
    id: 3,
    title: "Attack on Titan",
    url: "Mock URL AOT"
  },
]

describe('tests in GifGrid component', () => {
  const category = 'One Punch';
  const useGifMockLoading = (loadingStatus) => {
    return useFetchGifs.mockReturnValue({
      images: gifs,
      isLoading: loadingStatus
    })
  };

  test('should match snapshot when loading is true', () => {
    useGifMockLoading(true)

    const { container } = render(<GifGrid category={category} />)

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when loading is false', () => {
    useGifMockLoading(false)

    const { container } = render(<GifGrid category={category} />)

    expect(container).toMatchSnapshot();
  });

  test('should show loading at first instance', () => {
    useGifMockLoading(true);

    render(<GifGrid category={category} />);

    expect(screen.getByText('Gifs cargando...'));
  });

  test('should show content if is not loading', () => {
    useGifMockLoading(false);

    render(<GifGrid category={category} />);

    expect(screen.getByRole("heading", { level: 3 }).innerHTML).toContain(category)
  })
});
