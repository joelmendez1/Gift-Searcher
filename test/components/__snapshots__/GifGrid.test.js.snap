// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in GifGrid component should match snapshot when loading is false 1`] = `
<div>
  <h3>
    One Punch
  </h3>
  <div
    class="card-grid"
  >
    <div
      class="card"
    >
      <img
        alt="One Punch"
        src="Mock URL OP"
      />
      <p>
        One Punch
      </p>
    </div>
    <div
      class="card"
    >
      <img
        alt="Dragon Ball"
        src="Mock URL DB"
      />
      <p>
        Dragon Ball
      </p>
    </div>
    <div
      class="card"
    >
      <img
        alt="Attack on Titan"
        src="Mock URL AOT"
      />
      <p>
        Attack on Titan
      </p>
    </div>
  </div>
</div>
`;

exports[`tests in GifGrid component should match snapshot when loading is true 1`] = `
<div>
  <h2>
    Gifs cargando...
  </h2>
</div>
`;
